Test cases for turbulence model variants

suargi opened the pull request:
## Pull request description
This pull request introduces a series of regression cases for the Spalart-Allmaras turbulence model variations and corrections:

- Baseline
- Negative
- Edwards
- Compressibility
- Compressibility and Edwards
- QCR2000
 
~~The residuals on the python scripts are wrong. These values are temporary and will be updated once the regression tests are run on the Travis CI system.~~
Since this pull request does not introduce any code modification, the residuals on the python scripts have been updated on commit https://github.com/su2code/SU2/pull/1418/commits/1a1661f6cf51be9cdc3c642a7d0424016c7ea3fc with the Travis Cl (sim_vals (computed)) values.

## Related Work
These tests will be further used to verify the correct implementation of #1413 

- [X] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [X] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

suargi commented:
As stated in #1384 some hybrid parallel AD regression tests failed, more in precisely, the `discadj_incomp_cylinder`.
The regression test visc_cone has also failed with `ERROR: The code was not able to get to the "Begin solver" section.` for both serial_regression.py and parallel_regression.py. I do not know the reason.

WallyMaier commented:
@suargi An update to the TestCases repo renamed a folder for the visc_cone.  I have a PR open to fix the probleml

suargi commented:
> @suargi An update to the TestCases repo renamed a folder for the visc_cone. I have a PR open to fix the problem ---> Should be fixed now.
Thanks @WallyMaier. Now serial and parallel regression test work smoothly

pcarruscag commented on TestCases/turbulence_models/sa/rae2822/turb_SA_BSL_RAE2822.cfg:
```diff
@@ -0,0 +1,202 @@
+%
+% Specify turbulent model (NONE, SA, SA_NEG, SST)
+KIND_TURB_MODEL= SA
+%
+% Mathematical problem (DIRECT, CONTINUOUS_ADJOINT)
+MATH_PROBLEM= DIRECT
+%
+% Restart solution (NO, YES)
+RESTART_SOL= NO
```
```suggestion

```

pcarruscag commented on TestCases/turbulence_models/sa/rae2822/turb_SA_BSL_RAE2822.cfg:
```diff
@@ -0,0 +1,202 @@
+%                                                                              %
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
+%
+% Physical governing equations (EULER, NAVIER_STOKES,
+%                               WAVE_EQUATION, HEAT_EQUATION, FEM_ELASTICITY,
+%                               POISSON_EQUATION)                        
```
```suggestion
```

pcarruscag commented on TestCases/turbulence_models/sa/rae2822/turb_SA_BSL_RAE2822.cfg:
```diff
@@ -0,0 +1,202 @@
+%
+% Adaptive CFL number (NO, YES)
+CFL_ADAPT= NO
+%
+% Parameters of the adaptive CFL number (factor down, factor up, CFL min value,
+%                                        CFL max value )
+CFL_ADAPT_PARAM= ( 1.5, 0.5, 1.0, 100.0 )
```
```suggestion
```

pcarruscag commented on TestCases/turbulence_models/sa/rae2822/turb_SA_BSL_RAE2822.cfg:
```diff
@@ -0,0 +1,202 @@
+%                              TURKEL_PREC, MSW)
+CONV_NUM_METHOD_FLOW= JST
+%
+% Monotonic Upwind Scheme for Conservation Laws (TVD) in the flow equations.
+%           Required for 2nd order upwind schemes (NO, YES)
+MUSCL_FLOW= YES
+%
+% Slope limiter (VENKATAKRISHNAN, MINMOD)
+SLOPE_LIMITER_FLOW= VENKATAKRISHNAN
+%
+% Coefficient for the limiter (smooth regions)
+VENKAT_LIMITER_COEFF= 0.03
+%
```
```suggestion
```

pcarruscag reviewed (commented):
Thanks for this :+1: 
Please strip all the comments and non relevant options from the configs.

suargi commented:
> Thanks for this +1 Please strip all the comments and non relevant options from the configs.

Thank you Pedro for all your suggestions.
I have reverted the serial_regression.py file and updated the configuration files. This work is done in 84c9584d430e23ea62b4371cad3d0a96d9717cdb.

pcarruscag commented on TestCases/turbulence_models/sa/rae2822/turb_SA_COMP_EDW_RAE2822.cfg:
```diff
@@ -0,0 +1,202 @@
+% Case description: Transonic simulation RAE2822 (RANS)                        %
+% Author: Francisco Palacios                                                   %
+% Institution: Stanford University                                             %
+% Date: 5/15/2013                                                              %
```
We should just delete all that info from regression tests, if they are regressions then we know they are up to date with the code.

suargi replied:
Done it in 6bf7b7dd73d98faa9d847d19107505280b3f8433

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -620,6 +620,82 @@ def main():
+    turbmod_sa_bsl_rae2822.cfg_file  = "turb_SA_BSL_RAE2822.cfg"
+    turbmod_sa_bsl_rae2822.test_iter = 20
+    turbmod_sa_bsl_rae2822.test_vals = [-2.004689, 0.742306, 0.497308, -5.265793, 0.809463, 0.062016]
+    turbmod_sa_bsl_rae2822.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    turbmod_sa_bsl_rae2822.su2_exec  = "mpirun -n 2 SU2_CFD"
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -620,6 +620,82 @@ def main():
+    turbmod_sa_neg_rae2822.cfg_file  = "turb_SA_NEG_RAE2822.cfg"
+    turbmod_sa_neg_rae2822.test_iter = 20
+    turbmod_sa_neg_rae2822.test_vals = [-2.004688, 0.742320, 0.497307, -5.265640, 0.809478, 0.061986]
+    turbmod_sa_neg_rae2822.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    turbmod_sa_neg_rae2822.su2_exec  = "mpirun -n 2 SU2_CFD"
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -620,6 +620,82 @@ def main():
+    turbmod_sa_comp_rae2822.cfg_file  = "turb_SA_COMP_RAE2822.cfg"
+    turbmod_sa_comp_rae2822.test_iter = 20
+    turbmod_sa_comp_rae2822.test_vals = [-2.004687, 0.742304, 0.497309, -5.266081, 0.809467, 0.062029]
+    turbmod_sa_comp_rae2822.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    turbmod_sa_comp_rae2822.su2_exec  = "mpirun -n 2 SU2_CFD"
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -620,6 +620,82 @@ def main():
+    turbmod_sa_edw_rae2822.cfg_file  = "turb_SA_EDW_RAE2822.cfg"
+    turbmod_sa_edw_rae2822.test_iter = 20
+    turbmod_sa_edw_rae2822.test_vals = [-2.004687, 0.742306, 0.497310, -5.290769, 0.809485, 0.062036]
+    turbmod_sa_edw_rae2822.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    turbmod_sa_edw_rae2822.su2_exec  = "mpirun -n 2 SU2_CFD"
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -620,6 +620,82 @@ def main():
+    turbmod_sa_comp_edw_rae2822.cfg_file  = "turb_SA_COMP_EDW_RAE2822.cfg"
+    turbmod_sa_comp_edw_rae2822.test_iter = 20
+    turbmod_sa_comp_edw_rae2822.test_vals = [-2.004687, 0.742306, 0.497310, -5.290769, 0.809485, 0.062036]
+    turbmod_sa_comp_edw_rae2822.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    turbmod_sa_comp_edw_rae2822.su2_exec  = "mpirun -n 2 SU2_CFD"
```

pcarruscag commented on TestCases/parallel_regression.py:
```diff
@@ -620,6 +620,82 @@ def main():
+    turbmod_sa_qcr_rae2822.cfg_file  = "turb_SA_QCR_RAE2822.cfg"
+    turbmod_sa_qcr_rae2822.test_iter = 20
+    turbmod_sa_qcr_rae2822.test_vals = [-2.004793, 0.742353, 0.497315, -5.265974, 0.807841, 0.062027]
+    turbmod_sa_qcr_rae2822.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    turbmod_sa_qcr_rae2822.su2_exec  = "mpirun -n 2 SU2_CFD"
```
